src/main.rs

```diff
@@ -70,6 +67,77 @@ impl BFInterp {
             MetadataOperand => todo!(),
         }
     }
+    
+    fn load(&mut self, op: &Operand, dest: &llvm_ir::name::Name) {
+        match op {
+            LocalOperand { name, .. } => {
+                self.mem.insert(dest.clone(), *self.mem.get(name).unwrap());
+                self.mem.remove(name);
+            },
+            ConstantOperand(_) => todo!(),
+            MetadataOperand => todo!(),
+        }
+    }
+
+    fn call_func(&mut self, call: &Call) { 
+        let arg;    
+        match &call.arguments[1].0 {
+            LocalOperand { name, .. } => {
+                arg = self.mem.get(name).unwrap();
+            },
+            ConstantOperand(_) => todo!(),
+            MetadataOperand => todo!(),
+        }
+        
+        match &call.function {
+            Left(_) => todo!(),
+            Right(op) => {
+                match op {
+                    LocalOperand { .. } => todo!(),
+                    ConstantOperand(con_op) => {
+                        match con_op.as_ref() {
+                            Constant::GlobalReference { name, .. } => {
+                                match name {
+                                    Name(n) => {
+                                        if n.as_str() == "printf" {
+                                            println!("{}", arg);
+                                        }
+                                        // match self.find_symbol(n.as_str()) {
+                                        //     Some(c_void) => {
+                                        //         //println!("{:?}", c_void);
+                                        //         let func_pointer = c_void.cast::<fn()>();
+                                        //     },
+                                        //     None => todo!(),
+                                        // }
+                                    },
+                                    Number(_) => todo!(),
+                                }
+                            },
+                            _ => todo!(),
+                        }
+                    },
+                    MetadataOperand => todo!(),
+                }
+            },
+        }
+    }
+
+    fn find_symbol(&mut self, sym: &str) -> Option<*mut c_void> {
+        use std::ffi::CString;
+
+        let sym_arg = CString::new(sym).unwrap();
+        let addr = unsafe { dlsym(RTLD_DEFAULT, sym_arg.into_raw()) };
+
+        if addr == 0 as *mut c_void {
+            None
+        } else {
+            Some(addr)
+        }
+    }
+
+    fn exe_func(&mut self, func: fn()) {
+
```

Comment:
Do we need a `todo!()` here? Or at least a `// FIXME`  comment?

Reply:
Similar to the above, I think we can just remove this.